_posts/2022-06-22-domain-management-for-multitenant-app-in-aws.markdown

```diff
@@ -0,0 +1,62 @@
+
+
+Multitenancy is a challenging problem as you have to consider your application's security, optimization, monitoring, and high availability. Each aspect of implementation could be a real challenge when it grows. For a part like domain management, you don't think it will be a problem as you will only require to do most work during the initial setup. 
+
+However, it can be problematic along the way for a task such as SSL certificate renewal. If you have each certificate for each client's subdomain and each has a different expiration date, you need to manage them manually. Furthermore, for a task like setting up unique subdomains for new clients, what are the measures you take to ensure you will not impact your domains for other clients when there are changes to your current resources. When your number of clients grow, how do you have clear segregation of billing cost for each client? Lastly and most importantly, what is the disaster recovery plan for your instances, and how confident can you bring the application back online with minimal downtime? These are some overhead tasks we need to consider, and often most of the instances mentioned above are handled manually. 
+
+It might have more business value for you to focus on maintaining and developing the application, but managing the domain is critical in keeping the application live and accessible.
+
+### Solution
+
+The diagram below is the solution to how you can typically start building it, but it will soon become challenging when it grows.
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureSingleAccount.png)
+
+Generally, it is a good practice to segregate the production environment from the development environment, which should also apply to domain management. This approach has no segregation between environments and hosts all domain names within a single location. It increases the risk of impacting all tenants when there are service disruptions. 
+
+The architecture diagram below shows how to manage domains for all your tenants in multiple environments.  
+
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureDiagram.png)
+
+AWS Route53 domain management provides the ability to segregate the management of domain names between accounts. DNS is a public-facing element you would want to minimize downtime as it will directly impact your revenue or reputation. Having segregation between accounts, especially for non-production and production, is key to reducing the risk of damage to production domains. It is also beneficial for segregating billing costs; hence, you will get better insights. 
```

Comment:
> AWS Route53
This is the first time you have introduced AWS/Route53.  

* Is this article intended to be focussed on AWS?
* Could it apply for other services (and AWS is just your example)?

Depending on your intention for the above, be clear to identify the parts that are specific to an AWS implementation and which bits are more general.  For example, account management/structure is particular to AWS - you may manage it differently with other providers.

Not sure if you want to cover it in this article, but do you want to make an analogy with different hosting options / benefits - e.g. cloud services (like AWS) vs on-premise.  The way you look to manage domains may be different for each - creating new domains may be difficult in an on-prem environment.